### PR TITLE
[runtimes:crio] switch from parsing output in table format to json

### PR DESCRIPTION
**Pull request by npinaeva**
crictl added a field to the `crictl ps` output and that broke parsing - now we use json output format that (hopefully) won't change in the future

Signed-off-by: Nadia Pinaeva <npinaeva@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/policies/runtimes/crio.py [hunk 1]
```diff
@@ -47,13 +49,21 @@ def get_images(self):
...
+                    # takes the form (repository:tag, image_id)
+                    if len(image["repoTags"]) > 0:
+                        for repo_tag in image["repoTags"]:
+                            images.append((repo_tag, image["id"]))
+                    else:
+                        if len(image["repoDigests"]) == 0:
+                            image_name = "<none>"
+                        else:
+                            image_name = image["repoDigests"][0].split("@")[0]
+                        images.append((image_name + ":<none>", image["id"]))
```
Comment (TurboTurtle) [inline sos/policies/runtimes/crio.py:66]: I don't think adding `:<none>` here as part of the image name is the right move. That would require plugins to either use that specifically (self.add_cmd_output('foobar', container=`mycontainer:<none>`), or make their relevant container calls all use globs (`self.add_cmd_output('foobar', container=`mycontainer*`) which will probably add to problems down the line.

File: sos/policies/runtimes/crio.py [hunk 2]
```diff
@@ -29,14 +30,15 @@ def get_containers(self, get_all=False):
...
-            if out['status'] == 0:
-                for ent in out['output'].splitlines()[1:]:
-                    ent = ent.split()
+            if out["status"] == 0:
+                out_json = json.loads(out["output"])
+                for container in out_json["containers"]:
                     # takes the form (container_id, container_name)
-                    containers.append((ent[0], ent[-3]))
+                    containers.append(
+                        (container["id"], container["metadata"]["name"]))
```
Comment (TurboTurtle) [inline sos/policies/runtimes/crio.py:41]: Ack.

File: sos/policies/runtimes/crio.py (hunk 1 above)
Comment (npinaeva) [reply sos/policies/runtimes/crio.py:66]: so here I was trying to reproduce the output of table format for `crictl images` (also to be compatible with docker output), so adding none is what `crictl` does https://github.com/kubernetes-sigs/cri-tools/blob/ccb94be8f9c365f243d0c5c65fa7c26993d6094c/cmd/crictl/image.go#L522, image name can also be none https://github.com/kubernetes-sigs/cri-tools/blob/ccb94be8f9c365f243d0c5c65fa7c26993d6094c/cmd/crictl/image.go#L542
docker for command https://github.com/sosreport/sos/blob/main/sos/policies/runtimes/__init__.py#L122-L123 will return e.g.

```
# docker images --format '{{lower .Repository}}:{{lower .Tag}} {{lower .ID}}'
<none>:<none> dbfda3a8826c
quay.io/npinaeva/ovn-k:<none> a5851f835871

```
and for `self.add_cmd_output` we need container name and not image name - actually I don't think images are used in sos right now, just added that for completeness

Comment (TurboTurtle) [reply sos/policies/runtimes/crio.py:66]: Yep, I went cross eyed there and was mixing images with IDs.

**Pull request merged**